source/1_engine/rex_engine/src/diagnostics/assert.cpp

```diff
@@ -0,0 +1,85 @@
+#include "rex_engine/diagnostics/assert.h"
+
+#include "rex_engine/debug_types.h"
+#include "rex_engine/diagnostics/logging/log_macros.h"
+#include "rex_engine/diagnostics/stacktrace.h"
+
+namespace rex
+{
+  DECLARE_LOG_CATEGORY_EXTERN(LogAssert, LogVerbosity::Log);
+  DEFINE_LOG_CATEGORY(LogAssert);
+
+  DebugVector<AssertContext>& contexts()
+  {
+    thread_local DebugVector<AssertContext> contexts;
```

Comment:
Why does this need to be thread local?

Reply:
because I didn't want to introduce locks into the system.
a thread owns what it's running and the context it's running in.
so when a thread asserts it doesn't care about the contexts on a different thread (at least not right now, this might change in the future)